scripts/article1_post_draw.gml

```diff
@@ -38,110 +45,98 @@ if (x_dist == 0) {
     start_x = left_x;
     target_arc_length = chain_length / chain_segments;
 
+    end_iter = chain_segments - 1;
     for (i = 0; i < chain_segments; i++) {
-        x_guess = start_x + x_dist / chain_segments;
```

Comment:
this is the major optimization. previously, the arclength_with_start_target function would calculate the base arclength at x_start every iteration which was wasteful. now we just calculate it once and keep using it for this i-iteration. another improvement was the initial guess. making the initial guess the start_x value actually worked really well so our first iteration of newton's method is actually done outside the loop. since the arclength will be 0 between two values that are the same, we just use the difference in desired arclength for the output of the zeroing function and call the derivative regularly. inside the j-loop we reuse the start_arc but recalculate new_arc with the new guess and apply newton's method as usual